binderhub/repoproviders.py

```diff
@@ -882,7 +881,7 @@ def get_resolved_ref(self):
                              "'GistRepoProvider.allow_secret_gist = True'")
 
         all_versions = [e['version'] for e in ref_info['history']]
-        if (len(self.unresolved_ref) == 0) or (self.unresolved_ref == 'master'):
+        if self.unresolved_ref in {"", "HEAD", "master"}:
```

Comment:
gist was the only provider that didn't support HEAD because it's the only one where we resolve ref names in the provider instead of calling out to git/github/etc. I believe if we used `ls-remote` as in the git provider, this would not have needed any changes.